### ISSUE TITLE
[dv] tl_reg_adapater::bus2reg converts PutPartialData host writes to UVM_READ reg op

### ISSUE THREAD

**Issue by dkabely-ot** [lowRISC/opentitan#8068]
`tl_reg_adapter::bus2reg`, which converts bus transactions from TL-UL host/device to a `uvm_reg_bus_op` only converts bus transactions to `UVM_WRITE uvm_reg_bus_op` if `bus_item.a_opcode == PutFullData`.

However, `PutPartialData` is also a write request on the TL-UL bus, but it seems to be converted into a `UVM_READ uvm_reg_bus_op`.
https://github.com/lowRISC/opentitan/blob/d9dc048680d89bd630ffae673103509e28017709/hw/dv/sv/tl_agent/tl_reg_adapter.sv#L85

Shouldn't `PutPartialData` also be considered as an opcode which will be regarded as a write transaction and be converted by the adapter into a `UVM_WRITE uvm_reg_bus_op`?

In other words, shouldn't the condition above change into:
`rw.kind    = bus_rsp.is_write() ? UVM_WRITE : UVM_READ;`

**Comment by rswarbrick** [comment]
I think you're probably right (although I don't think we ever use `PutPartialData` in the OpenTitan system, which is why we haven't run into problems). Some quick git archaeology gets me back to the initial import.

@sriyerg: Can you think of anything I'm missing, or shall we switch this over?

**Comment by sriyerg** [comment]
Hi @dkabely-ot (and @rswarbrick), your suggested fix seems right to me. Please feel free to create a PR to address it.

**Comment by dkabely-ot** [comment]
@sriyerg Great. Will do.
